### PR TITLE
Feature/toggl diff sidekiq

### PR DESCRIPTION
**Pull request by siepet**
It does toggl diff in the background job and stores in the cache until it gets refreshed by a user.

### CODE REVIEW

File: config/config.yml [hunk 1]
```diff
@@ -52,6 +52,7 @@ test:
     email:
       account_using_instruction: dummy
     groups_whitelist: [group1@netguru.pl]
+  toggl_token: 'abcdefg'
```
Comment (lubieniebieski) [inline config/config.yml]: it's in `sec_config.yml.sample` - that should be enough - we will be notified when it's not overridden

File: spec/jobs/toggl/diff_worker_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe TogglWorkers::DiffWorker do
+  include_context 'data_guru'
+  include_context 'toggl_api'
+
+  before(:each) do
+    allow(TogglIntegration::Api).to receive(:new).and_return(toggl_api)
```
Comment (lubieniebieski) [inline spec/jobs/toggl/diff_worker_spec.rb]: couldn't you stub those methods on the instance?

File: app/jobs/toggl/base.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+module TogglWorkers
+  class Base < ActiveJob::Base
+    private
+
+    def data_guru
+      @data_guru ||= DataGuru::Client.new
+    end
+
+    def user_repo
```
Comment (lubieniebieski) [inline app/jobs/toggl/base.rb]: this method doesn't match the convention for other methods, it should be named `userS_repo`, right?

Comment (siepet) [inline app/jobs/toggl/base.rb]: I was doing `user_repo` everywhere, so I did `user_repo` here as well.

File: config/config.yml (hunk 1 above)
Comment (siepet) [inline config/config.yml]: Yes, but rspec would fail due to no token in `AppConfig`.

File: spec/jobs/toggl/diff_worker_spec.rb (hunk 2 above)
Comment (siepet) [inline spec/jobs/toggl/diff_worker_spec.rb]: Could you explain more what do you mean, please?

File: config/config.yml (hunk 1 above)
Comment (lubieniebieski) [inline config/config.yml]: yes, you're right, my bad

**Pull request merged**